Add serializer for struct Cookie, struct VideoFrameMetadata, struct NavigationPreloadState

chirags27 opened the pull request:
#### 4cc516b4855a16144a814fea1d9f02a32b8e0259
<pre>
Add serializer for struct Cookie, struct VideoFrameMetadata, struct NavigationPreloadState

rdar://problem/101179936

Reviewed by Alex Christensen.

* Source/WebCore/platform/Cookie.h:
(WebCore::Cookie::encode const): Deleted.
(WebCore::Cookie::decode): Deleted.
* Source/WebCore/platform/VideoFrameMetadata.h:
(WebCore::VideoFrameMetadata::encode const): Deleted.
(WebCore::VideoFrameMetadata::decode): Deleted.
* Source/WebCore/platform/WebCorePersistentCoders.cpp:
(WTF::Persistence::Coder&lt;WebCore::NavigationPreloadState&gt;::encode):
(WTF::Persistence::Coder&lt;WebCore::NavigationPreloadState&gt;::decode):
* Source/WebCore/workers/service/NavigationPreloadState.h:
(WebCore::NavigationPreloadState::encode const): Deleted.
(WebCore::NavigationPreloadState::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/255698@main">https://commits.webkit.org/255698@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5c2150477f6de6a195ae300ac42e325e5bb60fdb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93333 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2530 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23980 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103014 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/163345 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/97335 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2537 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30840 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/85707 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/99120 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99001 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/1775 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79790 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/28734 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/83688 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83449 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/71796 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37222 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/17318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35048 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18565 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3943 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38922 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41093 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40854 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/37785 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/platform/Cookie.h:
```diff
@@ -95,6 +92,24 @@ struct Cookie {
     enum class SameSitePolicy { None, Lax, Strict };
     SameSitePolicy sameSite { SameSitePolicy::None };
+
+    Cookie(String _name, String _value, String _domain, String _path, double _created, std::optional<double> _expires, bool _httpOnly, bool _secure, bool _session, String _comment, URL _commentURL, Vector<uint16_t> _ports, SameSitePolicy _sameSite) 
```
WebKit style prefers no _ at the beginning of parameter names.

achristensen07 replied:
Also a lot of these parameters should be r-values.  The strings and the vector for sure.
String&& name, String&& value, ... Vector<uint16_t>&&...
Then WTFMove below.

achristensen07 commented on Source/WebCore/platform/Cookie.h:
```diff
@@ -95,6 +92,24 @@ struct Cookie {
+        , ports(_ports)
+        , sameSite(_sameSite)
+    {
+
```
Please remove this whitespace.

achristensen07 commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -954,4 +954,51 @@ struct WebCore::AudioConfiguration {
- };
+};
+
+[Nested] enum class WebCore::Cookie::SameSitePolicy : int {
```
Could we instead make the underlying type a uint8_t?

chirags27 committed:
Add serializer for struct Cookie, struct VideoFrameMetadata, struct NavigationPreloadState

rdar://problem/101179936

Reviewed by Alex Christensen.

* Source/WebCore/platform/Cookie.h:
(WebCore::Cookie::encode const): Deleted.
(WebCore::Cookie::decode): Deleted.
* Source/WebCore/platform/VideoFrameMetadata.h:
(WebCore::VideoFrameMetadata::encode const): Deleted.
(WebCore::VideoFrameMetadata::decode): Deleted.
* Source/WebCore/platform/WebCorePersistentCoders.cpp:
(WTF::Persistence::Coder<WebCore::NavigationPreloadState>::encode):
(WTF::Persistence::Coder<WebCore::NavigationPreloadState>::decode):
* Source/WebCore/workers/service/NavigationPreloadState.h:
(WebCore::NavigationPreloadState::encode const): Deleted.
(WebCore::NavigationPreloadState::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/255698@main
